x-pack/plugins/transform/public/app/sections/transform_management/components/action_delete/use_delete_action.tsx

```diff
@@ -54,22 +61,37 @@ export const useDeleteAction = (forceDisable: boolean) => {
       userCanDeleteIndex && userCanDeleteDataView && indexPatternExists && deleteIndexPattern;
     // if we are deleting multiple transforms, then force delete all if at least one item has failed
     // else, force delete only when the item user picks has failed
-    const forceDelete = isBulkAction
+    const needForceDelete = isBulkAction
       ? shouldForceDelete
-      : items[0] && items[0] && items[0].stats.state === TRANSFORM_STATE.FAILED;
-
-    deleteTransforms({
-      transformsInfo: items.map((i) => ({
-        id: i.config.id,
-        state: i.stats.state,
-      })),
-      deleteDestIndex: shouldDeleteDestIndex,
-      deleteDestIndexPattern: shouldDeleteDestIndexPattern,
-      forceDelete,
-    });
+      : items[0] && items[0].stats?.state === TRANSFORM_STATE.FAILED;
+
+    deleteTransforms(
+      // If transform task doesn't have any corresponding config
+      // we won't know what the destination index or data view would be
+      // and should be force deleted
+      forceDelete && items[0].config === undefined
+        ? {
+            transformsInfo: items.map((i) => ({
+              id: i.id,
+              state: TRANSFORM_STATE.FAILED,
+            })),
+            deleteDestIndex: false,
+            deleteDestIndexPattern: false,
+            forceDelete: true,
+          }
+        : {
+            transformsInfo: items.map((i) => ({
+              id: i.id ?? i.config?.id,
+              state: i.stats?.state as TransformState,
```

Comment:
is it safe to cast the type here?

Reply:
I ended up removing the Delete modal and no longer need to do this :)